_pytest/deprecated.py

```diff
@@ -22,3 +27,6 @@
 GETFUNCARGVALUE = "use of getfuncargvalue is deprecated, use getfixturevalue"
 
 RESULT_LOG = '--result-log is deprecated and scheduled for removal in pytest 4.0'
+
+MARK_INFO_ATTRIBUTE = RemovedInPytest4_0Warning(
+    "Markinfo attributes are deprecated, please iterate the mark Collection")
```

Comment:
markinfo objects are iterables, atm the iteration yields markinfo objects as wel

@hpk42 i wonder if we can make it return the new mark type instead